consensus/ledger_validator/ledger_validator.c

```diff
@@ -261,8 +261,8 @@ retcode_t iota_consensus_ledger_validator_update_snapshot(
                                  .dynamic = 0};
   *has_snapshot = false;
 
-  if ((ret = iota_tangle_transaction_load(lv->tangle, TRANSACTION_FIELD_HASH,
-                                          &milestone_hash, &pack)) != RC_OK) {
+  if ((ret = iota_tangle_transaction_load_for_traversal(
+           lv->tangle, milestone->hash, &pack)) != RC_OK) {
```

Comment:
We only need snapshot + solid here, it would make sense to only load metadata

Reply:
you will need branch+trunk

Reply:
you are correct, confused it with do_func of traversals